### PR TITLE
Add support for Python 3.11

### PR DESCRIPTION
**Pull request by graingert**
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest, macos-latest]
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```
Comment (fjetter) [inline .github/workflows/tests.yaml:26]: Maybe we should exclude one osx build since they are slowing us down significantly.

Comment (fjetter) [reply .github/workflows/tests.yaml:26]: IIUC the environment is identical to 3.10. I suggest to skip OSX+3.10 then

**Review by crusaderky — APPROVED** [review]

**Review by jrbourbeau — COMMENTED** [review]
Thanks @graingert!

**Comment by consideRatio** [comment]
- Should https://github.com/dask/distributed/issues/6785 be closed now that this is merged?

**Comment by jrbourbeau** [comment]
I think so -- thanks for following up here @consideRatio

**Pull request merged**